### ISSUE TITLE
Three-column Component Group View for Local Section

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#4339]
This task is automatically imported from the old Task Issue Board and it was originally created by Adam Obuchowicz.
Original issue is [here](https://www.pivotaltracker.com/story/show/181414466).
<hr />

## Why

**As** Enso IDE developer
**I want** to see the three-column component group list without header
**So that** I can use it in the Local Scope Section.

## Acceptance Criteria
```gherkin
Scenario: 
GIVEN I open the debug scene with Component Groups
WHEN I look at it
THEN I see three-column view looking like the list "local scope" section
AND the items are ordered from left to right, from bottom to top.

```

## Notes

1. A three-column CGV should be composed of three `ListView` components placed side-by-side. These `Listview`s do not need headers, and they should share a single background for all of them (or, rather, background should be a separate component, and ListViews should have their background disabled).
2. A special `ModelProvider` should be implemented to provide all three `ListView`s from a single source. This `ModelProvider` should be parametrized with the ListView index (left, middle, or right) it is used with. The implementation should sort the entries alphabetically and distribute them among three `ListView`s evenly, in the left-to-right and bottom-to-top directions. 
```
The following (sorted) list of entries: [A, B, C, D, E, F, G, H, I] should be distributed this way:
Left | Middle | Right
G    | H      | I
D    | E      | F
A    | B      | C
```
3. Any "regular" CGV code should be abstracted out in such a way, that it can be reused by the "three-column" CGV, if necessary. This includes keyboard and mouse navigation.
4. In case there are no list entries to display, the View should display "No local variables" label instead of `ListView`s. 
5. ~~In case the visual height of the View is changing (the entries count is changing) - the View should be resized smoothly with an animation.~~ We (IB and AO) removed such requirement as it poses significant implementation difficulties.

## Blockers:
[Component Group implemented](https://www.pivotaltracker.com/story/show/Component Group implemented) resolved

## Comments:
**Ilya Bogdanov** reports a new **STANDUP** for the last Friday (2022-04-01):

**Progress:** technical design for three-column CGV is written. It should be finished by 2022-04-04. (Enso Bot - Apr 4, 2022)
<hr />
**Technical Design** review set to **pass** (Ilya Bogdanov - Apr 13, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-04-13):

**Summary:** There is **15 days** delay in implementation of the [Three-column Component Group View for Local Section  (#181414466) task](https://www.pivotaltracker.com/story/show/181414466).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** the task is started after tech design, so delay is not real (Enso Bot - Apr 14, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-04-13):

**Progress:** started the implementation of a new task, three-column component group. Made an initial visualisation stub, started working on ModelProvider wrapper to split entries into three lists. It should be finished by 2022-04-19.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181414466). continue working on three-column CG (I need to choose a better name, lol), also need to make a QA review of MC's task. (Enso Bot - Apr 14, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for today (2022-04-14):

**Progress:** Implemented a major part of the task: ModelProvider, visual layout of items, selected item filtering. It should be finished by 2022-04-19.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181414466). continue implementation, mostly visual improvements. (Enso Bot - Apr 14, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for the last Friday (2022-04-15):

**Progress:** working on dynamic entries count change, improved demo scene to test it and spotted a GPU buffers leakage in our implementation of list view. Investigated this issue and reported it. It should be finished by 2022-04-19.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181414466). finishing the implementation and polishing the code, also finishing scroll area task. (Enso Bot - Apr 18, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-04-18):

**Progress:** implemented dynamic resizing and "no items" label that is displayed when no items are provided. Also finished refactoring in Scroll Area task It should be finished by 2022-04-19.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181414466). refactoring and polishing the code (Enso Bot - Apr 19, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-04-19):

**Summary:** There is **2 days** delay in implementation of the [Three-column Component Group View for Local Section  (#181414466) task](https://www.pivotaltracker.com/story/show/181414466).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** polishing code for PR took a bit more time than I thought + items selection doesn't look correct when the selection moves from one column to another. This needs to be fixed. While preparing a PR, I faced an issue with our build system and also an interconnected one with enso-formatter. These issues need additional investigation. (Enso Bot - Apr 20, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-04-19):

**Progress:** was polishing the code to prepare a final PR. While testing I was stopped by broken `./run dist` command. Investigation took about an hour, resulted in a fix, but later I discovered another issue with enso-formatter. It should be finished by 2022-04-21.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181414466). fix item selection animation when moving the focus from one column to another, prepare PR if possible. (Enso Bot - Apr 20, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for today (2022-04-21):

**Summary:** There is **1 day** delay in implementation of the [Three-column Component Group View for Local Section  (#181414466) task](https://www.pivotaltracker.com/story/show/181414466).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** I was working on another task, was blocked by broken electron-builder + made adjustments to make selection animation look nice. (Enso Bot - Apr 21, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for today (2022-04-21):

**Progress:** implemented smooth animation when selecting items from different columns, prepared a PR and recorded a screencast. It should be finished by 2022-04-22. (Enso Bot - Apr 21, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 BLOCKER** for the last Friday (2022-04-22):

**Summary:** The [Three-column Component Group View for Local Section  (#181414466) task](https://www.pivotaltracker.com/story/show/181414466) is blocked by the [#181725016 task](https://www.pivotaltracker.com/story/show/181725016).

I don't know how to fill blocker size. Blocked by the implementation of a better abstraction for selecting items in list views. (Enso Bot - Apr 25, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-05-04):

**Summary:** There is **14 days** delay in implementation of the [Three-column Component Group View for Local Section  (#181414466) task](https://www.pivotaltracker.com/story/show/181414466).
It **will cause 0 day delay** for the delivery of this weekly plan.

6 days are weekends/holidays

**Delay Cause:** The rest is spent working on other tasks while this one was blocked + some additional delay for selection implementation. (Enso Bot - May 5, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-05-04):

**Progress:** I rebased the PR onto develop, added a support for new selection box, and was polishing the code and fixing minor bugs for the rest of the day. Attended a weekly team meeting. There are a few things that needs to be fixed: initial position of the selection; items slightly move up/down when reaching the start/end of the list; selection box should be hidden when no items are visible (so some API should be added for that). It should be finished by 2022-05-06.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181414466). fixing these minor issues, polishing the code, returning to the second task (Enso Bot - May 5, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-05-05):

**Progress:** Yesterday I was finishing the implementation, polishing the code and preparing a PR. Fixed the position of the selection box when the count of entries in the list changes, implemented hidding the selection when no entries are present. Addressing the review comments by WD. It should be finished by 2022-05-06.

**Next Day:** Next day I will be working on the [##181725003 task](https://www.pivotaltracker.com/story/show/181725003). returning to the second implementation task (Enso Bot - May 6, 2022)
<hr />
**Acceptance/QA** review set to **pass** (Adam Obuchowicz - May 10, 2022)
<hr />
